experiments/compositions/composition/tests/testclient/client.go

```diff
@@ -219,6 +238,118 @@ func (c *Client) MustExist(objs []*unstructured.Unstructured, timeout time.Durat
 	}
 }
 
+// MustHaveCondition - validate if the .status.conditions[] has the condition
+func (c *Client) MustHaveCondition(obj *unstructured.Unstructured, condition *metav1.Condition, timeout time.Duration) {
+	c.T.Helper()
+
+	toMatch := ExtractGVKNN(obj)
+	c.T.Logf("Checking condition %q present in object %q on cluster %q", condition.Type, toMatch, c)
+
+	unmatched := []*unstructured.Unstructured{obj}
+	errMissing := fmt.Errorf("condition missing")
+	errMismatch := fmt.Errorf("condition mismatch")
+
+	retryFrequency := getFrequency(c.T, opDuration, timeout)
+	err := retry.OnError(retryFrequency, func(err error) bool {
+		return apierrors.IsNotFound(err) || errors.Is(err, errMissing) || errors.Is(err, errMismatch)
```

Comment:
Oh ... now I get it!  Personally I would just always retry, we wouldn't expect too many surprises here.  You can always t.Errorf inside the retry block if you get something really unexpected.

Reply:
I remember vaguely that for some errors ( IsNotFound ), we can break early. Otherwise we would keep retrying till  the timeout. Makes tests faster.